src/Scientist/Scientist.cs

```diff
@@ -1,38 +1,75 @@
-﻿using GitHub.Internals;
-using System;
+﻿using System;
 using System.Threading.Tasks;
+using GitHub.Internals;
 
 namespace GitHub
 {
     /// <summary>
     /// A class for carefully refactoring critical paths. Use <see cref="Scientist"/> 
     /// </summary>
-    public static class Scientist
+    public class Scientist : IScientist
     {
-        static Func<Task<bool>> _enabled = () => Task.FromResult(true);
+        static readonly Task<bool> EnabledTask = Task.FromResult(true);
+        static readonly Lazy<Scientist> _sharedScientist = new Lazy<Scientist>(CreateSharedInstance);
+        static Func<Task<bool>> _enabled = () => EnabledTask;
+        static IResultPublisher _sharedPublisher = new InMemoryResultPublisher();
+        readonly IResultPublisher _resultPublisher;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Scientist"/> class
+        /// using the specified <see cref="IResultPublisher"/>.
+        /// </summary>
+        public Scientist(IResultPublisher resultPublisher)
+        {
+            if (resultPublisher == null)
+                throw new ArgumentNullException(nameof(resultPublisher), "A result publisher must be specified");
```

Comment:
Here's an opportunity to use the new C# throw expressions to make the code even simpler.

```csharp
_resultPublisher = resultPublisher
  ?? throw new ArgumentNullException(nameof(resultPublisher), "A result publisher must be specified");
```

Reply:
Changed.